lib/cisco_node_utils/itd_service.rb

```diff
@@ -0,0 +1,518 @@
+      @get_args = @set_args = keys
+    end
+
+    # rubocop:disable Style/AccessorMethodName
+    def set_args_keys(hash={})
+      set_args_keys_default
+      @set_args = @get_args.merge!(hash) unless hash.empty?
+    end
+
+    # extract value of property from load-balance
+    def extract_value(prop, prefix=nil)
+      prefix = prop if prefix.nil?
+      lb_match = lb_get
+
+      # matching lb not found
+      return nil if lb_match.nil? # no matching lb found
+
+      # property not defined for matching lb
+      return nil unless lb_match.names.include?(prop)
+
+      # extract and return value that follows prefix + <space>
+      regexp = Regexp.new("#{Regexp.escape(prefix)} (?<extracted>.*)")
+      value_match = regexp.match(lb_match[prop])
+      return nil if value_match.nil?
+      value_match[:extracted]
+    end
+
+    # prepend property name prefix/keyword to value
+    def attach_prefix(val, prop, prefix=nil)
+      prefix = prop.to_s if prefix.nil?
+      @set_args[prop] = val.to_s.empty? ? val : "#{prefix} #{val}"
+    end
+
+    def access_list
+      config_get('itd_service', 'access_list', @get_args)
+    end
+
+    def access_list=(val)
+      if val
+        @set_args[:state] = ''
+        @set_args[:al] = val
+        config_set('itd_service', 'access_list', @set_args)
+      else
+        @set_args[:state] = 'no'
+        @set_args[:al] = access_list
+        config_set('itd_service', 'access_list', @set_args) if
+        access_list
+      end
+      set_args_keys_default
+    end
+
+    def default_access_list
+      config_get_default('itd_service', 'access_list')
+    end
+
+    def device_group
+      config_get('itd_service', 'device_group', @get_args)
+    end
+
+    def device_group=(val)
+      if val
+        @set_args[:state] = ''
+        @set_args[:dg] = val
+        config_set('itd_service', 'device_group', @set_args)
+      else
+        @set_args[:state] = 'no'
+        @set_args[:dg] = device_group
+        config_set('itd_service', 'device_group', @set_args) if
+        device_group
+      end
+      set_args_keys_default
+    end
+
+    def default_device_group
+      config_get_default('itd_service', 'device_group')
+    end
+
+    def exclude_access_list
+      config_get('itd_service', 'exclude_access_list', @get_args)
+    end
+
+    def exclude_access_list=(val)
+      if val
+        @set_args[:state] = ''
+        @set_args[:al] = val
+        config_set('itd_service', 'exclude_access_list', @set_args)
+      else
+        @set_args[:state] = 'no'
+        @set_args[:al] = exclude_access_list
+        config_set('itd_service', 'exclude_access_list', @set_args) if
+        exclude_access_list
+      end
+      set_args_keys_default
+    end
+
+    def default_exclude_access_list
+      config_get_default('itd_service', 'exclude_access_list')
+    end
+
+    def failaction
+      config_get('itd_service', 'failaction', @get_args)
+    end
+
+    def failaction=(state)
```

Comment:
For the method and yaml prop name can we use `fail_action` instead?  Easier on the eyes :)